### ISSUE TITLE
Wall code modification + re-integration

### ISSUE THREAD

**Issue by FliiFe** [Apostolique/Agar.io-bot#596]
## <s>Wall code reintegration</s>
<s>As some may have noticed, Murat Lahur, before causing trouble to all of us, suggested to figure out the map borders. This is not a bad idea, but not really usefull as of now. However, we need the wall code in order not to be cornered. What I was thinking is that, we can just un-comment out the wall code, and it would (theoretically) work ! </s>
## <s>Why ?</s>
<s>Because the *fake* border agar.io gives are always centered on the blob, the only way you can reach one of the *fake* border of the map is by approaching the **real** ones. Thus, we can forget the *real* borders and ignore them.</s>

## What does the wall code do ?

The wall code is here to prevent the player from being cornered.
Basically, if there is no enemies around, or there are enemies but you are not in their danger zone, the wall code doesn't do anything.
However, when you get in a "danger zone", the bot will try to run away both the enemy and the wall. 
You may have noticed that while watching the bot, it is simbolized by pink-ish semi-circles around the cell.

## Wall code modification - Then, reintegration.

Above is explained why it should be added back.
But it should be modified so that the walls are detected from a smaller distance. (Please ask if this is unclear)

**Comment by ErmiyaEskandary** [comment]
In overall, removing the wall code makes the bot better everywhere else. 


> Because the fake border agar.io gives are always centered on the blob, the only way you can reach one of the fake border of the map is by approaching the real ones. Thus, we can forget the real borders and ignore them.

Don't fully understand this

**Comment by FliiFe** [comment]
The wall code removal doesn't make the bot better, it just removes an unuseful piece of the bot (we thought it was unuseful, at least).

So, I'll explain step by step.

The blob is **always** centered in the fake borders (the restricted map).
This is true, exept when you are close to a side/corner, when one or of the side will be the real borders.

That accepted, the blob will **only** be able to reach the border (fake or real) if it is close to a real border. So the wall code would activate itself only when this happens

Hope this is clearer

**Comment by ErmiyaEskandary** [comment]
Ah ...

**Comment by ErmiyaEskandary** [comment]
That's what I was thinking yesterday - but the location won't work as it's fake borders. 
But the wall code would work as it would only get activated when we are close to the **real** border ? Are we on the same page ?

**Comment by FliiFe** [comment]
Yes we are. Coordinates won't work, and there is a little we can do about this. But the wall code should work correctly if I'm not wrong

**Comment by ErmiyaEskandary** [comment]
Please make a commit to `dev` uncommenting the wall code - I agree with you

**Comment by ErmiyaEskandary** [comment]
Modified the wall code so that the walls are detected from a smaller distance. Testing...

**Comment by ErmiyaEskandary** [comment]
Initial testing successful. Will be added soon.

**Comment by Drflash55** [comment]
Huh... I always wondered about the problems with the wall code. What's important about it? Does it help with usually escaping incoming cells that are trying to corner you?

**Comment by ErmiyaEskandary** [comment]
That's what I know - helps with escaping incoming cells that are trying to corner you.

**Comment by FliiFe** [comment]
@Drflash55 Added clarification

**Comment by ErmiyaEskandary** [comment]
Will be added soon to the master branch.
